.github/actions/publish-image/action.yml

```diff
@@ -0,0 +1,57 @@
+name: "Publish Image to domain"
+description: "Publish Image to quay.io or dockerhub or another domain - only publish the version on master - elsewhere version and branch"
+
+inputs:
+  domain:
+    description: 'domain to publish image to'
+    required: true
+    type: string
+  username:
+    description: 'login username'
+    required: true
+    type: string
+  password:
+    description: 'login password'
+    required: true
+    type: string
+  dry-run:
+    description: dry run flag
+    required: true
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get docker image tag name
+      shell: bash
+      run: |
+        if [[ $BRANCH_NAME == "master" ]]; then
+            TAG_VERSION=$(grep -m1 version package.json | awk '{ print $2 }' | sed 's/[", ]//g')
+        else
+            TAG_VERSION=$BRANCH_NAME-${{ github.run_id }}
+        fi;
+        echo "TAG_VERSION=$TAG_VERSION" >> $GITHUB_ENV
+    - name: Publish image
+      shell: bash
+      run: |
+        DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+        cd $DIR/../../../
+
+        npm ci && npm run build.release
+
+        echo "Running the docker with tag" $TAG_VERSION
+        DOCKER_PROJECT_ARGS="PROJECT_NAME=content-ce"
+        DOCKER_REPOSITORY="${{ inputs.domain }}/${{ github.repository }}"
+
+        # Publish Image to quay.io or dockerhub or another domain - only publish the version on master - elsewhere version and branch
+        if [[ $BRANCH_NAME == "master" ]]; then
+          echo "npx @alfresco/adf-cli docker-publish --loginCheck --loginUsername '${{ inputs.username }}' --loginPassword '${{ inputs.password }}' --loginRepo '${{ inputs.domain }}' --dockerRepo '$DOCKER_REPOSITORY' --buildArgs  $DOCKER_PROJECT_ARGS  --dockerTags '$TAG_VERSION' "
+          if [[ ${{ inputs.dry-run }} == false ]]; then
+            npx       @alfresco/adf-cli docker-publish --loginCheck --loginUsername "${{ inputs.username }}" --loginPassword "${{ inputs.password }}" --loginRepo "${{ inputs.domain }}" --dockerRepo "$DOCKER_REPOSITORY" --buildArgs "$DOCKER_PROJECT_ARGS" --dockerTags "$TAG_VERSION" --pathProject "$(pwd)"
+          fi;
+        else
+          echo "npx @alfresco/adf-cli docker-publish --loginCheck --loginUsername '${{ inputs.username }}' --loginPassword '${{ inputs.password }}' --loginRepo '${{ inputs.domain }}' --dockerRepo '$DOCKER_REPOSITORY' --buildArgs  $DOCKER_PROJECT_ARGS  --dockerTags '$TAG_VERSION,$BRANCH_NAME' "
+          if [[ ${{ inputs.dry-run }} == false ]]; then
+            npx       @alfresco/adf-cli docker-publish --loginCheck --loginUsername "${{ inputs.username }}" --loginPassword "${{ inputs.password }}" --loginRepo "${{ inputs.domain }}" --dockerRepo "$DOCKER_REPOSITORY" --buildArgs "$DOCKER_PROJECT_ARGS" --dockerTags "$TAG_VERSION,$BRANCH_NAME" --pathProject "$(pwd)"
+          fi;
+        fi;
```

Comment:
generally speaking, when you have more than a few lines it's better to move to a dedicated .sh file and just call it from here, so you can have proper hilighting, indentation and optionally runs shellcheck within your editor/IDE

Reply:
Sure, moved this action to dedicated .sh